File: backend/services/activity.py
"""Activity log helper."""
from datetime import datetime, timezone
from typing import Optional
from core.database import db


async def log_activity(user_id: str, user_name: str, action: str, details: Optional[str] = None) -> None:
    await db.activity_logs.insert_one({
        "user_id": user_id,
        "user_name": user_name,
        "action": action,
        "details": details,
        "timestamp": datetime.now(timezone.utc),
    })

File: backend-node/src/services/activity.ts
/**
 * Activity log helper.
 * Equivalent of Python's services/activity.py
 */
import { db } from '../core/database';

export const logActivity = async (
  userId: string,
  userName: string,
  action: string,
  details?: string
): Promise<void> => {
  await db.activity_logs().insertOne({
    user_id: userId,
    user_name: userName,
    action,
    details: details ?? null,
    timestamp: new Date(),
  });
};
